src/data_management/wms_layers/WmsLayerForm.tsx

```diff
@@ -0,0 +1,357 @@
+import React, { useState } from 'react';
+import { RouteComponentProps, withRouter} from 'react-router';
+import {  useSelector, connect } from 'react-redux';
+import { ExplainSideColumn } from '../../components/ExplainSideColumn';
+import { CheckBox } from './../../form/CheckBox';
+import { TextArea } from './../../form/TextArea';
+import { TextInput } from './../../form/TextInput';
+import { SubmitButton } from '../../form/SubmitButton';
+import { CancelButton } from '../../form/CancelButton';
+import { SelectBox } from '../../form/SelectBox';
+import { SlushBucket } from '../../form/SlushBucket';
+import { AccessModifier } from '../../form/AccessModifier';
+import { useForm, Values } from '../../form/useForm';
+import { minLength,} from '../../form/validators';
+import wmsIcon from "../../images/wms@3x.svg";
+import {
+  getDatasets,
+  getOrganisations,
+  getSelectedOrganisation,
+  getSupplierIds,
+} from '../../reducers';
+import { addNotification } from './../../actions';
+import formStyles from './../../styles/Forms.module.css';
+import SpatialBoundsField from "../../forms/SpatialBoundsField";
+import MinMaxZoomField, {MinMax} from '../../components/MinMaxZoomField';
+import { WmsLayerReceivedFromApi, wmsLayerReceivedFromApiToForm, WmsLayerFormType, wmsLayerGetDefaultFormValues, wmsLayerFormToFormSendToApi} from '../../types/WmsLayerType';
+
+interface Props {
+  currentWmsLayer?: WmsLayerReceivedFromApi, 
+};
+
+interface PropsFromDispatch {
+  addNotification: (message: string | number, timeout: number) => void,
+};
+
+const WmsLayerForm: React.FC<Props & PropsFromDispatch & RouteComponentProps> = (props) => {
+  const { currentWmsLayer} = props;
+
+  const [geoserverError, setGeoserverError,] = useState(false)
+  const organisationsToSharedWith = useSelector(getOrganisations).availableForRasterSharedWith;
+  const supplierIds = useSelector(getSupplierIds);
+  const organisations = useSelector(getOrganisations).available;
+  const selectedOrganisation = useSelector(getSelectedOrganisation);
+  const datasets = useSelector(getDatasets).available;
+
+  const initialValues: WmsLayerFormType = currentWmsLayer ? wmsLayerReceivedFromApiToForm(currentWmsLayer) : wmsLayerGetDefaultFormValues(selectedOrganisation.uuid);
+  
+  const onSubmit = (values: Values) => {
+
+    // @ts-ignore
+    const wmsLayer = wmsLayerFormToFormSendToApi(values);
+    const url = "/api/v4/wmslayers/";
+
+     if (!currentWmsLayer) {
+        const opts = {
+          credentials: "same-origin",
+          method: "POST",
+          headers: { "Content-Type": "application/json" },
+          body: JSON.stringify(wmsLayer),
+        }
+        // @ts-ignore
+        fetch(url, opts)
+        // @ts-ignore
+        .then((data:any) => {
```

Comment:
Maybe later, we can consider to have a wmsApi.ts file in the api directory that would do all the api calls like the rasterApi.

Reply:
We will probably soon get swagger files (similar to 3di backend). If they are ready we should gradually remove all these api call functions and type definitions wherever possible.